Firestore/core/src/nanopb/nanopb_util.h

```diff
@@ -104,6 +104,69 @@ T* _Nonnull MakeArray(pb_size_t count) {
   return static_cast<T*>(calloc(count, sizeof(T)));
 }
 
+template <typename T>
+T* _Nonnull ResizeArray(void* _Nullable ptr, pb_size_t count) {
+  return static_cast<T*>(realloc(ptr, count * sizeof(T)));
+}
+
+/**
+ * Initializes a repeated field with a list of values. Applies `converter` to
+ * each value before assigning.
+ */
+template <typename T, typename Iterator, typename Func>
+void SetRepeatedField(T** fields_array,
+                      pb_size_t* fields_count,
+                      Iterator first,
+                      Iterator last,
+                      Func converter) {
```

Comment:
Nit: you can pass `converter` by const reference. Since this is a template parameter, it could in theory stand for, say, a lambda with many captures.